nibabel/gifti/gifti.py

```diff
@@ -680,6 +680,43 @@ def get_arrays_from_intent(self, intent):
         it = intent_codes.code[intent]
         return [x for x in self.darrays if x.intent == it]
 
+    def agg_data(self, intent_code=None):
+        """
+        Retrun a numpy arrary of aggregated GiftiDataArray of the same intent code
+        or
+        Retrun GiftiDataArray in tuples for surface files
+
+        Parameters
+        ----------
+        intent_code : None, string, integer or tuple of string, optional
+            Intent code, or string describing code.
+            Accept tuple that contains multiple intents to specify the order.
+
+        Returns
+        -------
+        all_data : tuple of ndarray or ndarray
+            If the input is a tuple, the returned tuple will match the order.           
+        """
+
+        # Allow multiple intents to specify the order
+        # e.g., agg_data(('pointset', 'triangle')) ensures consistent order
+
+        if isinstance(intent_code, tuple):
+            return tuple(self.agg_data(intent_code=code) for code in intent_code)
+
+        darrays = self.darrays if intent_code is None else self.get_arrays_from_intent(intent_code)
+        all_data = tuple(da.data for da in darrays)
+        all_intent = tuple(intent_codes.niistring[da.intent] for da in darrays)
+
+        # Gifti files allows usually one or more data array of the same intent code
+        # surf.gii is a special case of having two data array of different intent code
+
+        if (self.numDA > 1 and all(el == all_intent[0] for el in all_intent)):
+            return np.column_stack(all_data)
```

Comment:
I'm not sure about this one. It's not clear to me that:

1) We can assume multiple arrays of the same intent will be stackable. The shapes might not work out. For example, suppose I have a GIFTI with multiple pointsets of different lengths. (I don't know why you'd want this, but it's definitely not ruled out by the standard.)
2) A user will prefer this stacking for any arbitrary intent. For instance, if I have a two-parameter statistical distribution like `NIFTI_INTENT_NORMAL`, I might get alternating columns `(mu, sigma, mu, sigma, ...)`. In that case, perhaps concatenating on a new axis would be preferred. The nice thing about time series is that the spec is pretty clear that each time step is a 1-D array. And if we see `NIFTI_INTENT_NONE`, all bets are off.

I'm hesitant to make concatenation contingent on the shapes working out, as the output type will then be an even more complex function of the input data than we're already proposing. I see a few options:

1) Try to `np.column_stack`, and if it fails, raise an error saying `agg_data` failed, so please munge the data arrays yourself.
2) `np.column_stack` for time series only (I didn't see any others that seem obvious candidates), `tuple` for everything else. This should be pretty safe.
3) Take option (2) by default, but allow it to be parameterized with an `aggregator` parameter:
    ```Python
    def agg_data(self, intent_code=None, aggregator=None):
        if aggregator is None:
            if intent_code == 'NIFTI_INTENT_TIME_SERIES':
                aggregator = np.column_stack
            else:
                aggregator = tuple
    ```
4. Require an aggregator parameter, using `tuple` by default. This would make time series a less special case, but would make the return type much more predictable.

I'm inclined toward (2), with an option to move to (3) if people actually want control over that. Not sure they will, as they can always do `np.column_stack(img.agg_data())`. WDYT?

Reply:
I wrote something but for got to hit reply.....

Anyway, the TLDR of my original comment was, these are great suggestions! I didn't think about those weird but totally legit use of gifti format. I agreed to go for 2 for now and move to 3. I think we will need more people to use these functions to know what is common for users. 
Option 1 is just cruel. Option 4 is going to be confusing for users that don't really know the data structure.